### ISSUE TITLE
Asynchrony between Carla and Autoware

### ISSUE THREAD

**Issue by Kim-mins** [evshary/autoware_carla_launch#5]
Hi @evshary
I'm currently running Autoware with Carla following the documentation of this repository (**branch: humble**)
I set everything correctly following the instruction, but when I set the goal position on RVIZ, it seems Autoware and Carla are not communicating well..
Here is the link to the video to show the situation:
https://drive.google.com/file/d/1bO2ef6-LXW9Zwt4_nTf0vuX-jLRm2gCa/view?usp=sharing

I also observed that, even though the car made a bit sharper turn, it eventually collides with the bridge on the right side (which means the car does not steer to the left)

I do not know about how to make a log file regarding Autoware or bridge, but I'll do my best to make the situation crystal clear.
I just need your help. Thank you in advance!

Here is my environment and running command:
- OS: Ubuntu 20.04
- Carla Version: 0.9.13
- I slightly modified the bridge script (options like `--sync`, `--autopilot`, `--pygame` and `--position`) and other commands are the same with that of the documentation.
```bash
#!/bin/bash
PYTHON_AGENT_PATH=${AUTOWARE_CARLA_ROOT}/external/zenoh_carla_bridge/carla_agent
parallel --verbose ::: "RUST_LOG=z=info ${AUTOWARE_CARLA_ROOT}/external/zenoh_carla_bridge/target/release/zenoh_carla_bridge" \
                       "poetry -C ${PYTHON_AGENT_PATH} run python3 ${PYTHON_AGENT_PATH}/main.py \
                        --host ${CARLA_SIMULATOR_IP} --rolename ${VEHICLE_NAME} --autopilot --sync --pygame --position $1"
```
I also tried after removing option `--sync`, but the situation was the same.

**Comment by evshary** [comment]
Can you remove `--autopilot` and `--sync`, then run again?
I don't face your problem by my side. Did this always happen or sometimes happen?

**Comment by evshary** [comment]
BTW, it would be helpful if you can give me your vehicle position. Then I can try to reproduce your issue.
Also, you can try to slow down Carla simulation time by modifying the value here (100ms => 200ms) and see whether it can work.
https://github.com/evshary/zenoh_carla_bridge/blob/6333a10731bfaccea463e82a069e59c849e56ae5/src/main.rs#L68
The path: `external/zenoh_carla_bridge/src/main.rs`
Remember to rerun `make build_bridge` again.

**Comment by Kim-mins** [comment]
Thank you for the reply @evshary!
Actually, for the first time of the running, I did not suffer this issue. So it does not happen all the time, I guess.
For the reproduction, the vehicle spawning position is `{ x: 299.3700844574216, y: -132.33484190502503, z: -0.22391417467260943, 'roll': 0.0, 'pitch': 0.0, 'yaw': -9.1552734375e-05}`.
But I remember that I also fall into this issue even on other positions..
I'll try more with your suggestions, and I'll tell you if I could find any hint for the solution. Thank you!

**Comment by Kim-mins** [comment]
@evshary for the spawning point above, you can use this string for the input for running `main.py` in the bridge script:
`'299.3999938964844,133.2400360107422,0.29999998211860657,0.0,0.0,-9.1552734375e-05'`

**Comment by Kim-mins** [comment]
Oh sorry @evshary, I just come up with that I also slightly modified the script `./container/run-bridge-docker.sh` and `./container/run-autoware-docker.sh`.
For the modification, I copied the command generated after running those script(the command seems to be generated by rocker, and I copied the command printed after the string `Executeng command:` on CLI).
For those commands, I removed `--rm` and added `-d` and `--name ~~~`. As an example, `./container/run-bridge-docker.sh` is re-written as:
```
docker run -d -it --network host   --gpus all --privileged --name autoware_bridge -v /home/mins/minsu2/autoware-universe-humble/autoware_carla_launch:/home/mins/autoware_carla_launch  -e DISPLAY -e TERM   -e QT_X11_NO_MITSHM=1   -e XAUTHORITY=/tmp/.docker3gwgs97a.xauth -v /tmp/.docker3gwgs97a.xauth:/tmp/.docker3gwgs97a.xauth   -v /tmp/.X11-unix:/tmp/.X11-unix   -v /etc/localtime:/etc/localtime:ro  45523eff379c
```
so that I can reuse the container even after I exited from the container.

For each docker container, I ran the bridge and autoware by using `docker exec`, and I faced the situation above.
For the comparison, I ran the bridge and autoware following the instruction of this repository, and everything was OK.
So maybe this is **the setting error on my side**. Maybe I should run those strictly following your instruction.
Sorry for making you spend your valuable time.
I really appreciate to your help. Thank you.
